Client/core/CChat.cpp

```diff
@@ -699,6 +736,46 @@ bool CChat::CharacterKeyHandler(CGUIKeyEventArgs KeyboardArgs)
     return true;
 }
 
+bool CChat::KeyDownHandler(CGUIKeyEventArgs KeyboardArgs)
+{
+    // If we can take input
+    if (!CLocalGUI::GetSingleton().GetConsole()->IsVisible() && IsInputVisible())

```

Comment:
Maybe refactor this into `CChat::CanTakeInput()`? Also short-circuiting here might be nicer